### PR TITLE
Allow SQS Visibility Timeout to be an option

### PR DESCRIPTION
**Pull request by dokie**
To allow the consumer from SQS to potentially override the visibility timeout set on the queue, each message read from the queue can set it upon the read action.

I ultimately am looking at how additionally I could set this on each read at runtime to cater for a slow running consumer but without setting a very high value on the queue or in this option. This will allow a kind of "keep-alive" for a message being worked upon. This is facilitated via [this](https://docs.aws.amazon.com/AWSSimpleQueueService/latest/APIReference/API_ChangeMessageVisibility.html) call on the AWS API and I think this is supported on the `ExAws.SQS` client.

### CODE REVIEW

**Comment by msaraiva** [comment]
Hi @dokie! 

Nice work! Can you please run `mix format` and push it again?

Thanks!

**Comment by dokie** [comment]
Hi

I have ran it on my local machine and I always have the formatter run on file saves too so it seems not to change anything. I will try down versioning my Elixir version from 1.8 to 1.7 and see if that helps.

Mike

File: lib/broadway_sqs/ex_aws_client.ex [hunk 1]
```diff
@@ -97,8 +98,25 @@ defmodule BroadwaySQS.ExAwsClient do
   defp validate_option(:wait_time_seconds, value) when not is_integer(value) or value < 0,
     do: validation_error(:wait_time_seconds, "a non negative integer", value)
 
-  defp validate_option(:max_number_of_messages, value) when value not in 1..10,
-    do: validation_error(:max_number_of_messages, "a integer between 1 and 10", value)
+  defp validate_option(:max_number_of_messages, value)
+       when value not in 1..@max_num_messages_allowed_by_aws,
+       do:
```
Comment (josevalim) [inline lib/broadway_sqs/ex_aws_client.ex]: Given those are large, let's make them proper do/end blocks.

File: lib/broadway_sqs/ex_aws_client.ex [hunk 2]
```diff
@@ -97,8 +98,25 @@ defmodule BroadwaySQS.ExAwsClient do
...
+           :max_number_of_messages,
+           "an integer between 1 and #{@max_num_messages_allowed_by_aws}",
+           value
+         )
+
+  defp validate_option(:visibility_timeout, nil), do: {:ok, nil}
+
+  defp validate_option(:visibility_timeout, value)
+       when value not in 0..@max_visibility_timeout_allowed_by_aws_in_seconds,
+       do:
```
Comment (josevalim) [inline lib/broadway_sqs/ex_aws_client.ex]: Given those are large, let's make them proper do/end blocks.

**Comment by josevalim** [comment]
@dokie Thanks for reformatting!

@msaraiva we should update our .travis.yml to run on v1.8 and check for formatted only on v1.8. :)

**Comment by josevalim** [comment]
Also, should we update the docs? @msaraiva can you point to the proper place where that would be?

File: lib/broadway_sqs/ex_aws_client.ex (hunk 1 above)
Comment (dokie) [reply lib/broadway_sqs/ex_aws_client.ex]: Will do - no problem

File: lib/broadway_sqs/ex_aws_client.ex (hunk 2 above)
Comment (dokie) [reply lib/broadway_sqs/ex_aws_client.ex]: Will do - no problem

**Comment by msaraiva** [comment]
Looks good. Documentation has been already updated.

**Pull request merged**